### PR TITLE
Apply the new component linter rules to most components

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

Removes exceptions from tasks/components.py, and fixes existing components that do not match the new conventions.

### Motivation

Clean up components and make them consistent with the documentation.

### Describe how to test/QA your changes

N/A

### CODE REVIEW

**Review by davidor — APPROVED** [review]
:+1: for container-platform files.

File: comp/trace/compression/fx-zstd/fx.go [hunk 1]
```diff
@@ -3,8 +3,8 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-// Package fxzstd provides fx options for the compression component.
-package fxzstd
+// Package fx provides fx options for the compression component.
+package fx
```
Comment (GustavoCaso) [inline comp/trace/compression/fx-zstd/fx.go:7]: Looks like we have a discrepancy between the docs and the linter 😄 
https://datadoghq.dev/datadog-agent/components/creating-components/#the-fx-folders-compcompressionfx-zstdfxgo

Could we make sure that linters and documentation matches

**Review by rayz — APPROVED** [review]

File: comp/trace/compression/fx-zstd/fx.go (hunk 1 above)
Comment (dustmop) [reply comp/trace/compression/fx-zstd/fx.go:7]: PR up for documentation update: https://github.com/DataDog/datadog-agent/pull/28072

**Review by jmw51798 — APPROVED** [review]

**Review by songy23 — APPROVED** [review]
LGTM for OTel

**Review by ichinaski — APPROVED** [review]

File: comp/trace/agent/fx-mock/fx.go [hunk 2]
```diff
@@ -15,6 +16,8 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
+// team: agent-apm
```
Comment (GustavoCaso) [inline comp/trace/agent/fx-mock/fx.go:19]: Should the team comment by in the definition package?

**Review by GustavoCaso — APPROVED** [review]
LGTM

File: comp/trace/agent/fx-mock/fx.go (hunk 2 above)
Comment (dustmop) [reply comp/trace/agent/fx-mock/fx.go]: You're correct, they are unnecessary. Removed.

**Pull request merged**